socksio/utils.py

```diff
@@ -2,11 +2,15 @@
 import functools
 import socket
 import typing
+import re
 
 if typing.TYPE_CHECKING:
     from socksio.socks5 import SOCKS5AType  # pragma: nocover
 
 
+IP_V6_WITH_PORT_REGEX = re.compile(r"\[(?P<address>.+)\]:(?P<port>\d+)")
```

Comment:
Let's use `^\[(?P<address>[^\]]+)\]:(?P<port>\d+)$` instead of `.+` in `address` to get a quicker match failure on invalid addresses.